File: kumoh-code-2017/#2 설계/과제2 학생관리 프로그램/학생관리/학생관리/StudentList.cpp
#include "StudentList.h"

void StudentList::insertStudent(Student s)
{
	student_list[count] = s; // 생성자의 내용을 배열에 삽입할 수 있게 객체 생성

	count++; // 다음 공간에 삽입할 수 있게 증가
}

void StudentList::deleteStudent(int index)
{
	student_list[index] = student_list[index + 1]; // 전의 공간에 다음 공간을 덮어 씌워준다

	count--; // 다음 공간에 삽입할 수 있게 증가된 공간이 비어있기 때문에 입력된 내용까지만 보여주기 위해 감소
}

void StudentList::updateStudent(int index , Student s)
{
	student_list[index] = s; // 학생 정보 검색코드가 들어갈 수 있는 공간의 Index를 생성
}

File: kumoh-code-2017/Java/#1 기본/#1 과제/#1 java 과제/실습2/실습 2-1/StudentList.java
import java.util.Scanner;

public class StudentList
{
	Scanner sc = new Scanner(System.in);
	
	final int MAX_STUDENT = 100;
	
	private Student[] student_list = new Student[MAX_STUDENT];
	private int count;
	
	public void registerStudent()
	{
		System.out.println("학생 정보를 등록합니다.");
		
		System.out.print("학과 : ");
		String major = sc.next(); // 학과
		
		System.out.print("학년 : ");
		int grade = sc.nextInt(); // 학년
		
		System.out.print("학번 : ");
		int id = sc.nextInt(); // 학번
		
		System.out.print("이름 : ");
		String name = sc.next(); // 이름
		
		System.out.print("성별 : ");
		String sex = sc.next(); // 성별
		
		System.out.print("생일 : ");
		String birth = sc.next(); // 생일
		
		System.out.print("체중 : ");
		float weight = (float)sc.nextInt(); // 체중
		
		System.out.print("주소 : ");
		String address = sc.next(); // 주소
		
		Student s = new Student(major, grade, id, name, sex, birth, weight, address);
		
		student_list[count] = s;
		count++;
		
		System.out.println();
	}
	
	public void displayStudent()
	{
		System.out.println("학생 정보를 출력합니다.");
		
		for(int i = 0; i < count; i++)
		{
			System.out.println("학과 : " + student_list[i].getMajor());
			System.out.println("학년 : " + student_list[i].getGrade());
			System.out.println("학번 : " + student_list[i].getId());
			System.out.println("이름 : " + student_list[i].getName());
			System.out.println("성별 : " + student_list[i].getSex());
			System.out.println("생일 : " + student_list[i].getBirth());
			System.out.println("체중 : " + student_list[i].getWeight());
			System.out.println("주소 : " + student_list[i].getAddress());
			System.out.println();
		}
	}
}
